python_modules/dagster-graphql/dagster_graphql/schema/roots/mutation.py

```diff
@@ -329,6 +331,24 @@ def mutate(self, graphene_info: ResolveInfo, backfillId: str):
         return resume_partition_backfill(graphene_info, backfillId)
 
 
+class GrapheneAddDynamicPartitionMutation(graphene.Mutation):
+    """Adds a partition to a dynamic partition set."""
+
+    Output = graphene.NonNull(GrapheneAddDynamicPartitionResult)
+
+    class Arguments:
+        partitionsDefName = graphene.NonNull(graphene.String)
+        partitionKey = graphene.NonNull(graphene.String)
+
+    class Meta:
+        name = "AddDynamicPartitionMutation"
+
+    @capture_error
+    @require_permission_check(Permissions.EDIT_DYNAMIC_PARTITIONS)
+    def mutate(self, graphene_info: ResolveInfo, partitionsDefName: str, partitionKey: str):
+        return add_dynamic_partition(graphene_info, partitionsDefName, partitionKey)
```

Comment:
Hey gang! Hmm @clairelin135  I'm not sure that we expose the partition definition name to the UI -- it looks like this is a property of `DynamicPartitionsDefinition`, maybe we can expose it on `GraphenePartitionDefinition`? I don't think we have a way to retrieve this param currently!

Separately it'd actually be nice to have an ID of some sort on these -- I think the UI has to compare partition definitions for equality by comparing their typename and dimension types + names.